### ISSUE TITLE
Create an integration test to verify CreateSnapshot/CreateTableFromSnapshot work.

### ISSUE THREAD

**Issue by coryan** [GoogleCloudPlatform/google-cloud-cpp#1053]
We need an integration test for CreateSnapshot and CreateTableFromSnapshot. This should create a table, write some data to it, then create a snapshot, create a new table with the snapshot.

In production we should be able to check that the schema and contents match, but I do not know if that is possible using the emulator.

**Comment by manish-qlogic** [comment]
> but I do not know if that is possible using the emulator.

`snapshottable` does not work with emulator, it throws following error
`2018/09/02 20:20:59 Failed to create Snapshot: rpc error: code = Unavailable desc = transport is closing`

Emulator logs suggests SIGSEGV
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x7d0663]
```
so, submitted the code that runs against production but does not run against emulator.

**Comment by coryan** [comment]
This was fixed by #1079.
